bot_python_sdk/store.py

```diff
@@ -8,69 +9,93 @@
 _bot_public_key = 'storage/public.pem'
 
 
+LOCATION = 'Store'
+
+
 class Store:
     @staticmethod
     def set_actions(actions):
-        file = open(_actions_file_path, 'w')
-        file.write(json.dumps(actions))
-        file.close()
+        try:
+            with open(_actions_file_path, 'w') as outfile:
+                outfile..write(json.dumps(actions))
+        except IOError:
+            Logger.error(LOCATION, 'Unable to write in '+_actions_file_path+' path')
 
     @staticmethod
     def get_actions():
         if not os.path.isfile(_actions_file_path):
             return []
-        file = open(_actions_file_path, 'r')
-        actions = json.loads(file.read())
-        file.close()
-        return actions
+        try:
+            with open(_actions_file_path, 'r') as outfile:
+                actions = json.loads(outfile.read())
+                return actions
+        except IOError:
+            Logger.error(LOCATION, 'Unable to read in '+_actions_file_path+' path')
+            return None
+
 
     @staticmethod
     def get_last_triggered(action_id):
         if not os.path.isfile(_last_triggered_file_path):
             return None
-        file = open(_last_triggered_file_path)
-        data = json.loads(file.read())
-        file.close()
-        return data[action_id] if action_id in data.keys() else None
+        try:
+            with open(_last_triggered_file_path) as outfile:
+                data = json.loads(outfile.read())
+                return data[action_id] if action_id in data.keys() else None
+        except IOError:
+            Logger.error(LOCATION, 'Unable to read '+_last_triggered_file_path+' path')
+            return None
 
     @staticmethod
     def set_last_triggered(action_id, time):
         data = {}
         if os.path.isfile(_last_triggered_file_path):
-            file = open(_last_triggered_file_path, 'r')
-            data = json.loads(file.read())
-            file.close()
+            with open(_last_triggered_file_path, 'r') as outfile:
+                data = json.loads(outfile.read())
         data[action_id] = time
-        file = open(_last_triggered_file_path, 'w')
-        file.write(json.dumps(data))
-        file.close()
+        try:
+            with open(_last_triggered_file_path, 'w') as outfile:
+                outfile.write(json.dumps(data))
+        except IOError:
```

Comment:
I like that you log the errors here, but I don't agree with catching the error completely. So I think we should either raise the error again so the calling code can handle it, or create custom errors for this class and raise that, so we are very specific about what type of errors we raise.

Reply:
i have raise the error in except block